scripts/albs-1147.py

```diff
@@ -0,0 +1,231 @@
+        (r'kvm-rhel8.3', 'kvm-almalinux8.3'),
+        ('rhel-9', 'almalinux-9'),
+        ('rhel-8', 'almalinux-8'),
+        ('rhel9.2', 'almalinux9.2'),
+        ('rhel-8.5', 'almalinux-8.5'),
+        ('rhel 8.4', 'almalinux 8.4')
+    )
+    for pattern, repl in branding_patterns:
+        original_string = re.sub(pattern, repl, original_string)
+    return original_string
+
+
+def log_differences(original: str, new: str):
+    original_words = original.split()
+    new_words = new.split()
+
+    matcher = difflib.SequenceMatcher(None, original_words, new_words)
+
+    for tag, i1, i2, j1, j2 in matcher.get_opcodes():
+        if tag == 'replace':
+            original_change = ' '.join(original_words[i1:i2])
+            new_change = ' '.join(new_words[j1:j2])
+
+            start_context = max(0, i1 - 3)
+            end_context = min(len(original_words), i2 + 3)
+            original_context = ' '.join(original_words[start_context:i1]) \
+                               + ' 【' + original_change + '】 ' \
+                               + ' '.join(original_words[i2:end_context])
+
+            start_context = max(0, j1 - 3)
+            end_context = min(len(new_words), j2 + 3)
+            new_context = ' '.join(new_words[start_context:j1]) \
+                          + ' 【' + new_change + '】 ' \
+                          + ' '.join(new_words[j2:end_context])
+
+            log.info(f'... {original_context} ➔ {new_context}')
+
+
+async def main(write=False):
+    ignore_parts = [
+        'access.redhat.com',
+        'container-tools:rhel8',
+        'eclipse:rhel8',
+        'go-toolset:rhel8',
+        'jmc:rhel8',
+        'llvm-toolset:rhel8',
+        'rust-toolset:rhel8',
+        'virt:rhel',
+        'virt-devel:rhel',
+        'ansible-collection-redhat-rhel_mgmt',
+        'ansible-collection-AlmaLinux-rhel_mgmt',
+        'rhel-system-roles',
+        'lorax-templates-rhel',
+        'redhat-rpm-config',
+        'kmod-redhat-oracleasm',
+        'rhel8 stream'
+    ]
+    search_parts = [
+        'rhel',
+        'red hat',
+        'redhat'
+    ]
+    affected_records = {}
+
+    with get_pulp_db() as session:
+        result = session.execute(
+            select(UpdateRecord).where(
+                or_(
+                    *[UpdateRecord.description.ilike(f'%{part}%') for part in search_parts],
+                    *[UpdateRecord.title.ilike(f'%{part}%') for part in search_parts],
+                ),
+                not_(
+                    or_(
+                        *[UpdateRecord.description.ilike(f'%{part}%') for part in ignore_parts],
+                        *[UpdateRecord.title.ilike(f'%{part}%') for part in ignore_parts],
+                    )
+                )
+            )
+        )
+        records = result.scalars().all()
+
+        log.info(f'Found {len(records)} records in Pulp\'s \'rpm_updaterecord\' table.')
+        for record in records:
```

Comment:
I suggest to move logic for modifications in Pulp in one place, at the end of the script. Because right now it looks a bit weird. Also, you can fetch the data once and compile the list of repositories for publication while debranding